### ISSUE TITLE
Amazon searches still being tracked while Facebook Container enabled

### ISSUE THREAD

**Issue by khemion** [mozilla/contain-facebook#123]
- Is "Firefox will: Never remember history" in the Firefox Preferences/Options under "Privacy & Security > History" selected? Yes/No: **No**
- Are you using Firefox in a Private Window? Yes/No: **No**
- Can you see a grayed out but ticked Checkbox with the description "Enable Container Tabs" in the Firefox Preferences/Options under "Tabs"? Yes/No: **Yes**
- Facebook Container Version: **1.3.1**
- Operating System + Version: **Windows 8.1.1**
- Firefox Version: **59.0.2 (64-bit)**
- Other installed Add-ons + Version + Enabled/Disabled-Status: **Adblock 3.9.0, Enabled; BetterTTV 7.0, Enabled; FrankerFaceZ 4.0, Enabled; Honey 10.6.7, Enabled**

### Actual behavior
When scrolling Facebook, I am seeing sponsored posts that indicate Facebook is still tracking my browsing history, particularly my Amazon searches. The sponsored posts I am seeing are for Amazon products that I viewed over the weekend, despite having installed Facebook Container several days ago.

### Expected behavior
I wasn't sure exactly what to expect, but I was hoping that Facebook Container would help prevent Facebook from tracking my searches on other websites including Amazon.

### Notes
Is this considered normal, or is anybody else experiencing this issue? Could there be something that I'm doing wrong? Thank you for your help. :)

**Comment by stoically** [comment]
In general it's unfortunately still possible to track you because of [Browser fingerprinting](https://en.wikipedia.org/wiki/Device_fingerprint); Facebook Container separates storage (like cookies), but it can't really help against fingerprinting. Out of curiosity - did you use "Login with Facebook" on a Website before or after doing your amazon searches (in the same Browser session)?

**Comment by khemion** [comment]
Thanks for the response. :) I don't have any accounts connected with Facebook that I remember, and if I do then it's been at least a year since I logged into any.


This internet privacy rabbit hole is crazy! I'm a casual user but looking into this issue brings so many things to light. I wonder if there's another add-on or other means to keep Facebook from accessing my browser fingerprint?

**Comment by TanviHacks** [comment]
@khemion 
Are you using Facebook on Mobile as well?  If so, through the Facebook App, or a browser?

Are you using Amazon on Mobile?  if so, through the Amazon App, or a browser?

If you are using Facebook and Amazon both in a mobile browser, then that can explain this.  Unfortunately, I doubt you are :(

Facebook could use your fingerprint or your IP address to try and identify you.  I was hoping they weren't doing this type of sophisticated tracking, since it is really against the users desire, given these users have taken measures to stop the tracking.

Anyway, if you can provide a little more information on your setup, then maybe we can find another issue, other than fingerprinting and IP address correlation.

Note, there is a resistFingerprint Mode in Firefox.  You can set it through a hidden pref, but it breaks a few things.  The Tor browser sets it. (To turn it on, open a tab and type about:config.  Promise you will be careful.  Search for the pref privacy.resistFingerprinting and set it to true instead of false.  Or install this extension that just does that: https://addons.mozilla.org/en-US/firefox/addon/resist-fingerprinting/)

**Comment by khemion** [comment]
@TanviHacks Thanks! I fear my lack of knowledge and tendency to fat-finger would make working with Tor a bit too perilous, haha. I'll give the add-on a shot and see how it works.


I'm using Firefox on my desktop computer, and browse both Facebook and Amazon on the desktop site. Don't have any tablets, smart home "assistants" (like Alexa) or other devices. I do have the Amazon app on my smartphone but haven't used it in a long time. I did also have the Facebook and Messenger apps on my phone for a little while, but they creeped me out too much so I deleted them a couple months ago. Not sure what other info to provide, as I'm not tech savvy in this realm.


Thanks again for your help. :)

**Comment by khemion** [comment]
Update: as an experiment I downloaded the add-on, browsed Amazon for a bit (unique products I hadn't searched before), then opened Facebook and scrolled for awhile. Didn't see any sponsored posts with those Amazon products. So far so good!

**Comment by englehardt** [comment]
The advertisements could also be part of Facebook's custom audiences program: https://www.facebook.com/business/help/170456843145568. As an overview, the program lets retailers (such as Amazon) upload customer lists to advertise to them on Facebook. The customer list file can use information like your name, email address, phone number (see: https://www.facebook.com/business/help/606443329504150?helpref=faq_content).

It's possible Amazon uses custom audiences for retargeted ads on Facebook. There are two ways to check:
1. Each ad should have a ["Why am I seeing this ad"](https://mashable.com/2014/06/12/facebook-explains-ads/#xVTsoejdGmqL) option that will explain how it was targeted.
2. You can view the list of advertisers that have uploaded your personal information for ad targeting. Here's a [guide](https://mic.com/articles/168986/how-to-find-which-advertisers-have-your-contact-info-on-facebook#.FiKHb27Qg) on how to get to that page.

**Comment by TanviHacks** [comment]
@khemion note that the resistfingerprinting option can break things sometimes.  so if you see breakage, disable the resist fingerprinting addon temporarily.

You should also consider turning on Tracking Protection - https://support.mozilla.org/en-US/kb/tracking-protection

**Comment by khemion** [comment]
@englehardt Well, that explains it! I did some more digging in my Facebook ad preferences... I thought I'd cleaned things up before, but it turns out that Amazon was on the list of companies still allowed to share my contact/browsing information with Facebook (yeesh). Went ahead and turned that off and will keep an eye on those settings going forward.

Also, @TanviHacks - thank you, I will keep that in mind. :) I turned on Tracking Protection as soon as I installed Firefox, along with the automatic "do not track" requests (which is why I was initially baffled at how that, plus this container app, still wasn't enough to keep those creepy sponsored ads away). As an aside, it's crazy how many sites have that little shield icon pop up... why do companies seem to think that people will be okay with feeling spied on? I have nothing to hide, but it's still really unnerving.
